Okta.Xamarin/Okta.Xamarin/IOktaStateManager.cs

```diff
@@ -0,0 +1,48 @@
+﻿// <copyright file="OktaState.cs" company="Okta, Inc">
+// Copyright (c) 2019-present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Okta.Xamarin
+{
+    /// <summary>
+    /// State manager interface.
+    /// </summary>
+    public interface IOktaStateManager
+    {
+        string AccessToken { get; }
+
+        IOktaConfig Config { get; set; }
+
+        DateTime? Expires { get; }
+
+        string IdToken { get; }
+
+        bool IsAuthenticated { get; }
+
+        HttpResponseMessage LastApiResponse { get; set; }
+
+        string RefreshToken { get; }
+
+        string Scope { get; }
+
+        string TokenType { get; }
+
+        void Clear();
+
+        string GetToken(TokenType tokenType);
+
+        Task<ClaimsPrincipal> GetUserAsync(string authorizationServerId = "default");
```

Comment:
It feels like this method should be part of the client since it's not necessarily related to the state itself. In many other OIDC libraries, this method usually lives in the client. Thoughts?

Reply:
I through the same at first but I'm following the pattern established by the iOS Sdk where the state manager is the top level component that developers interact with; the Oidc client is used exclusively to handle the authentication and code exchange flow.